app/src/main/java/org/thoughtcrime/securesms/preferences/ClearAllDataDialog.kt

```diff
@@ -97,46 +115,52 @@ class ClearAllDataDialog : DialogFragment() {
         }
     }
 
-    private fun clearAllData(deleteNetworkMessages: Boolean) {
-        clearJob = lifecycleScope.launch(Dispatchers.IO) {
-            val previousStep = step
-            withContext(Dispatchers.Main) {
-                step = Steps.DELETING
+    private suspend fun performDeleteLocalDataOnlyStep() {
+        try {
+            ConfigurationMessageUtilities.forceSyncConfigurationNowIfNeeded(requireContext()).get()
+        } catch (e: Exception) {
+            Log.e(TAG, "Failed to force sync when deleting data", e)
+            withContext(Main) {
+                Toast.makeText(ApplicationContext.getInstance(requireContext()), R.string.errorUnknown, Toast.LENGTH_LONG).show()
+            }
+            return
+        }
+        val success = ApplicationContext.getInstance(context).clearAllData(false)
+        withContext(Main) { dismiss() }
+        if (!success) {
+            withContext(Main) {
+                Toast.makeText(ApplicationContext.getInstance(requireContext()), R.string.errorUnknown, Toast.LENGTH_LONG).show()
             }
+        }
+    }
 
-            if (!deleteNetworkMessages) {
-                try {
-                    ConfigurationMessageUtilities.forceSyncConfigurationNowIfNeeded(requireContext()).get()
-                } catch (e: Exception) {
-                    Log.e("Loki", "Failed to force sync", e)
-                }
-                ApplicationContext.getInstance(context).clearAllData(false)
-                withContext(Dispatchers.Main) {
-                    dismiss()
-                }
-            } else {
-                // finish
-                val result = try {
+    private fun clearAllData(deletionScope: DeletionScope) {
+
+        clearJob = lifecycleScope.launch(Dispatchers.IO) {
+            withContext(Main) { step = Steps.DELETING }
```

Comment:
may as well remove the scheduling and put this on Line 138, do it synchronously is safer, no?